Samples/2.0/ChatRoom/GrainImplementation/Channel.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using GrainInterfaces;
+using GrainInterfaces.Model;
+using Orleans;
+using Orleans.Streams;
+using Utils;
+
+namespace GrainImplementation
+{
+	public class Channel : Grain, IChannel
+	{
+		private readonly List<ChatMsg> messages = new List<ChatMsg>(100);
+		private readonly List<string> onlineMembers = new List<string>(10);
+
+		private IAsyncStream<ChatMsg> stream;
+
+		public override Task OnActivateAsync()
+		{
+			var streamProvider = GetStreamProvider(Constants.ChatRoomStreamProvider);
+
+		    stream = streamProvider.GetStream<ChatMsg>(Guid.NewGuid(), Constants.CharRoomStreamNameSpace);
+            return base.OnActivateAsync();
+		}
+
+		public async Task<Guid> Join(string nickname)
+		{
+			if (onlineMembers.Contains(nickname)) return Guid.Empty;
```

Comment:
It seems confusing to me to return a `Guid.Empty` here. I think it's fine for a second client for the same nickname to join the channel.